doc/user-guide/indexing.rst

```diff
@@ -536,6 +533,31 @@ __ https://docs.scipy.org/doc/numpy/user/basics.indexing.html#assigning-values-t
       da.isel(x=[0, 1, 2])[1] = -1
       da
 
+You can also assign values in all variables of a ``Dataset`` at once:
+
+.. ipython:: python
+
+    ds = xr.tutorial.open_dataset("eraint_uvz")
```

Comment:
can we construct the dataset manually? That way, the effect is easier to see:
```suggestion
    ds = xr.Dataset(
        {"u": (("latitude", "longitude", "level"), np.zeros(shape=(5, 2, 2)))},
        coords={
            "latitude": [10, 10.5, 11, 11.5, 12],
            "longitude": [10, 20],
            "level": [100, 200],
        },
    )
```
of course, that means we would have to update the labels below, and we might need one more variable with a `Dataset` on the right-hand side.

Reply:
Interesting — I rather liked loading the tutorial dataset rather than creating one specifically, since it's terser and to the extent many examples start doing this, they may become familiar.

@keewis would printing `ds` satisfy your concern? This isn't a strongly held view of mine, and I'm likely only seeing part of the context, so happy to be wrong here.

Reply:
I agree that the tutorial dataset is nice and it's better to use real-world examples rather than synthetic ones, but the effect of the operation has to be clearly visible. Also, for me an operation is always easier to understand if I can see all of the data: only with that I can try to figure out what it did, and if I can't see all the data I won't be able to verify that it didn't do anything else. This is not possible with the entire tutorial dataset (hence the suggestion), but I guess we could change this to
```suggestion
    ds = (
        xr.tutorial.open_dataset("eraint_uvz")
        .isel(latitude=slice(0, 5), longitude=slice(0, 5), level=200)
    )
```
such that we're always working with small enough data that it can be printed. I also like the use of `zeros` (we could use `xr.zeros_like` for this) because then the contrast between old and new is even higher, but if we're assigning constants that's not much of an issue.

Reply:
I think that's a great synthesis of the two options!

Reply:
I agree! In the new version I select a slice of the tutorial dataset and fill it with zeros.